cmd/gokeyless/initialize.go

```diff
@@ -71,19 +71,35 @@ func initAPICall(token *apiToken, csr string) ([]byte, error) {
 	}
 	defer resp.Body.Close()
 
-	apiResp := new(initAPIResponse)
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("certificate API returned an invalid response body for HTTP %d", resp.StatusCode)
+	}
+
+	apiResp := &initAPIResponse{}
+	if resp.StatusCode != http.StatusOK {
+		// Check if the returned body is JSON by attempting to unmarshal it into
+		// our response struct. If it fails, return the HTTP status code and raw
+		// body for debugging.
+		err := json.Unmarshal(bodyBytes, apiResp)
```

Comment:
Get rid of this `resp.StatusCode != http.StatusOK` check and `err := json.NewDecoder(resp.Body).Decode(apiResp)` below. You've already read the `resp.Body` into `bodyBytes`, so it can't be passed to the decoder later.

Just call 

```
if err := json.Unmarshal(bodyBytes, apiResp); err != nil {
	log.Debugf("invalid JSON response: %s", bodyBytes)
	return nil, fmt.Errorf("certificate API returned HTTP %d", resp.StatusCode)
}
if !apiResp.Success {		
 	errs, _ := json.Marshal(apiResp.Errors)		
 	return nil, fmt.Errorf("api call failed: %s", errs)		
 }
```